### PR TITLE
AsyncLoop and PeriodicTask tests

### PR DESCRIPTION
**Pull request by Neurosploit**
-Corrected repeateEvery typo in asyncloop.
-AsyncLoop/PeriodicTask tests
-Added stopping log message to Asyncloop and PeriodicTask classes.
-Added logging unit test base class
-Stopped parallel unit test because of instability of test runs see assembyinfo.

### CODE REVIEW

File: Stratis.Bitcoin.Tests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -33,4 +34,7 @@
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
 [assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.0.0.0")]
\ No newline at end of file
+[assembly: AssemblyFileVersion("1.0.0.0")]
+
+// disable parallel running of tests because of concurrency issues with the log class being static (see LogsTestBase).
+[assembly: CollectionBehavior(DisableTestParallelization = true, MaxParallelThreads = 1)]
```
Comment (dangershony) [inline Stratis.Bitcoin.Tests/Properties/AssemblyInfo.cs]: Ok this is a good intensive to instance the logging freamwork
Maybe add a todo to remember to enable this back? (next commit)

File: Stratis.Bitcoin/AsyncLoop.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+{
+    public interface IAsyncLoop
+    {
+        string Name { get; }
+
+        Task Run(TimeSpan? repeatEvery = default(TimeSpan?), TimeSpan? startAfter = default(TimeSpan?));
+        Task Run(CancellationToken cancellation, TimeSpan? repeatEvery = default(TimeSpan?), TimeSpan? startAfter = default(TimeSpan?));
+    }
+
+    public class AsyncLoop : IAsyncLoop
```
Comment (dangershony) [inline Stratis.Bitcoin/AsyncLoop.cs]: Now that its an instance there is a good case to add the CancellationProvider in the constructor.
And the method CancelleationToken override that.

File: Stratis.Bitcoin/BlockStore/BlockStoreLoop.cs [hunk 3]
```diff
@@ -98,13 +98,13 @@ public void Loop(CancellationToken cancellationToken)
...
 			// or downloads missing blocks then writing to store
 			AsyncLoop.Run("BlockStoreLoop.DownloadBlocks", async token =>
-			{
-				await DownloadAndStoreBlocks(cancellationToken);
-			},
-			cancellationToken,
-			repeateEvery: TimeSpans.Second,
-			startAfter: TimeSpans.FiveSeconds);
-		}
+			  {
```
Comment (dangershony) [inline Stratis.Bitcoin/BlockStore/BlockStoreLoop.cs]: Indentation

File: Stratis.Bitcoin.Tests/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (Neurosploit) [inline Stratis.Bitcoin.Tests/Properties/AssemblyInfo.cs]: Yes. I will do that.

File: Stratis.Bitcoin/AsyncLoop.cs (hunk 2 above)
Comment (Neurosploit) [inline Stratis.Bitcoin/AsyncLoop.cs]: I'm not sure what you refer to with cancellationprovider. I take it you mean cancellationtokensource?

**Pull request merged**